internal/storage/storage_handle.go

```diff
@@ -43,6 +43,17 @@ type StorageClientConfig struct {
 	HttpClientTimeout   time.Duration
 	MaxRetryDuration    time.Duration
 	RetryMultiplier     float64
+	UserAgent           string
+}
+
+type userAgentRoundTripper struct {
+	inner     http.RoundTripper
+	UserAgent string
+}
+
+func (ug *userAgentRoundTripper) RoundTrip(r *http.Request) (*http.Response, error) {
+	r.Header.Set("User-Agent", ug.UserAgent)
+	return ug.inner.RoundTrip(r)
 }
```

Comment:
should we move this to seperate file to keep it cleaner?

Reply:
fixed